### ISSUE TITLE
bun dependency

### ISSUE THREAD

**Issue by AVVS** [googleapis/nodejs-firestore#21]
Depends on an outdated library, which restricts node engine to 0.10.x

```
? Choose which packages to update. firebase-admin@5.4.0
info Installing "dependencies"...
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error bun@0.0.11: The engine "node" is incompatible with this module. Expected version "~0.10.12".
✨  Done in 2.31s.
Error: Found incompatible module
    at checkOne (/usr/local/Cellar/yarn/1.1.0/libexec/lib/cli.js:59750:11)
    at Object.check (/usr/local/Cellar/yarn/1.1.0/libexec/lib/cli.js:59756:5)
    at /usr/local/Cellar/yarn/1.1.0/libexec/lib/cli.js:22277:73
    at Generator.next (<anonymous>)
    at step (/usr/local/Cellar/yarn/1.1.0/libexec/lib/cli.js:92:30)
    at /usr/local/Cellar/yarn/1.1.0/libexec/lib/cli.js:103:13
    at <anonymous>
```

I didnt personally review the code of the library and it might be just fine, but there is an issue here one way or another

Thanks!

**Comment by jmdobry** [comment]
Looking at what you posted, it's not clear to me how your error is related to the `@google-cloud/firestore` NPM package.

**Comment by AVVS** [comment]
https://github.com/googleapis/nodejs-firestore/blob/master/package.json#L44 - this module (bun) - pins node engine to 0.10.x - https://github.com/naomik/bun/blob/master/package.json#L19, runtime of cloud functions is 4.x.x as far as I remember, local node is 8.x.x

yarn (package manager) verifies all dependencies and based on the field dims it incompatible with the current runtime (bun might be just fine, its just that note in the package.json)

I, of course, can setup the package manager to ignore such "warnings" and force it to install the dependency, but it's a problem one way or another that the module `@google-cloud/firestore` depends on wants node runtime to be `0.10.x`

**Comment by hiranya911** [comment]
Perhaps replace bun with merge2? 

https://www.npmjs.com/package/merge2

**Comment by Swivelgames** [comment]
I'm getting this error as well. `nodejs-firestore` is using the `bun` npm module which has a dependency on node engine `0.10.x` which is now breaking `firebase` implementations (among others).

```
error bun@0.0.11: The engine "node" is incompatible with this module. Expected version "~0.10.12".
error Found incompatible module
```

This just cropped up a few hours ago.

**Comment by Swivelgames** [comment]
@schmidt-sebastian @jmdobry `bun` was just updated to `0.0.12` and supports the latest version of the `node` engine 👍 :)

**Comment by lukesneeringer** [comment]
Fixed by #24.
